### ISSUE TITLE
Unable to ping ports 31400-31409 (open)

### ISSUE THREAD

**Issue by Sebastiaan512** [pi-node/instructions#75]
DHCP Reservation - router forwarding - firewall input rule

First - DHCP Reservation for my computer with the Mac address associated with my ethernet port.
Second - Through the router with the reserved IP I've done a port forwarding with the ports 31400-31409.
Third - Made a new input rule in windows following the instructions 31400-31409.

![Pi node](https://user-images.githubusercontent.com/59236513/109334998-ab1ba600-781e-11eb-9133-3083d22aa784.png)

Port forwarding w/  view of my available  router firewall and advanced options.
![T3200m](https://user-images.githubusercontent.com/59236513/109336630-eb7c2380-7820-11eb-8020-6f8ac0e91b6e.png)

**Comment by ihatejam** [comment]
fyi - your LAN IP address cannot be used by us outside of your network - so it is typically not necessary to mask it (but it doesn't hurt to be careful).
It is the router's public/WAN ip address that needs to be masked - this is the one that is typically directly reachable from outside - which I in fact suspect is not the case here and is the source of your grief...

The thing to check next is whether the ISP is doing their part - i.e. allowing port-forwarding. Something that could help you check this is:-

(1.) go here to check your port forwarding - it will autoreport your public IP address- chances are that it will fail - but it helps to sanity check things - https://www.ipfingerprints.com/portscan.php
(1.1.) are ports 31400-31409 reporting open? (I'm guessing not)
(1.2.) note down the public ip address that this site reports you to be using (do NOT post it here) - I'll call it <PUBLIC_IP>

(2) Check the config of your router to see what ip address it reports as being assigned by your ISP (do NOT post it here) - let's call it <WAN-IP> - note it down.

If <PUBLIC_IP> and <WAN-IP> are the same, then your setup/config is likely the issue (if the ISP is not blocking port-forwarding).

Now if <PUBLIC_IP> and <WAN-IP> are NOT the same, then either 
(A) you have an additional router (sometimes more than one) sitting between your main router (usually provided by your ISP) and the router you are configuring the forwarding on.
or
(B) your ISP does not provide "true" public IP addresses to its subscribers - so your port-forwarding efforts will have zero effect.

To recover from (A) a bit more work is needed - you need to forward those same ports from the ISP's router to the router connecting to your node. If there are multiple routers in between then each will have to be set up with forwarding those ports to the next router in the chain. You can also get around this with a VPN service (supporting port-forwarding) like Surfshark or NordVPN.
  
To recover from (B) with help from your ISP - call them up and explain that you want to do port forwarding. If they can't help - then either move to an ISP that can do it or to use a VPN service (supporting port-forwarding) like Surfshark or NordVPN.

**Comment by Sebastiaan512** [comment]
I really appreciate the detailed help THANK YOU!

> fyi - your LAN IP address cannot be used by us outside of your network - so it is typically not necessary to mask it (but it doesn't hurt to be careful). 

**Didn't know this**

> It is the router's public/WAN ip address that needs to be masked - this is the one that is typically directly reachable from outside - which I in fact suspect is not the case here and is the source of your grief...
> 
> The thing to check next is whether the ISP is doing their part - i.e. allowing port-forwarding. Something that could help you check this is:-
> 
> (1.) go here to check your port forwarding - it will autoreport your public IP address- chances are that it will fail - but it helps to sanity check things - https://www.ipfingerprints.com/portscan.php
> (1.1.) are ports 31400-31409 reporting open? (I'm guessing not) Host is up.

**PORT     -          STATE   -  SERVICE**

**31400/tcp        filtered     pace-licensed**

**others are "filtered" state with "unknown" service**

> (1.2.) note down the public ip address that this site reports you to be using (do NOT post it here) - I'll call it <PUBLIC_IP>
> 
> (2) Check the config of your router to see what ip address it reports as being assigned by your ISP (do NOT post it here) - let's call it - note it down.
> 
> If <PUBLIC_IP> and are the same, then your setup/config is likely the issue (if the ISP is not blocking port-forwarding).
> 
> Now if <PUBLIC_IP> and are NOT the same, then either
> (A) you have an additional router (sometimes more than one) sitting between your main router (usually provided by your ISP) and the router you are configuring the forwarding on.
> or
> (B) your ISP does not provide "true" public IP addresses to its subscribers - so your port-forwarding efforts will have zero effect.

**It seams to be different (Modem IP Address is what I'm looking at correct?)**
![settings](https://user-images.githubusercontent.com/59236513/109343531-c2609080-782a-11eb-9620-147f5eb5418f.png)

> 
> To recover from (A) a bit more work is needed - you need to forward those same ports from the ISP's router to the router connecting to your node. If there are multiple routers in between then each will have to be set up with forwarding those ports to the next router in the chain. You can also get around this with a VPN service (supporting port-forwarding) like Surfshark or NordVPN.
> 
> To recover from (B) with help from your ISP - call them up and explain that you want to do port forwarding. If they can't help - then either move to an ISP that can do it or to use a VPN service (supporting port-forwarding) like Surfshark or NordVPN.

**Ill call my ISP but I have the feeling that it wont make a difference based off forums where they're mentioned.**

**I have Nord, I didn't know it could do that. I'm guessing its the custom DNS or manually changing the "TCP/IPv4" to the** **NordVPN's DNS server. Otherwise on their FAQ port forwarding isn't integrated for safety.** 

**Once again thank you for replying**

**Comment by ihatejam** [comment]
Yep. The result you got from ipfingerprints site implies that the issue is with your ISP. This is confirmed by the mismatch between your modem ip address and what the outside world (ipfingerprints) thinks it should be. So other pi nodes really have no clear route to reach your pc. 
I have personally used surfshark vpn while traveling to keep my node working - no additional config was necessary besides the local windows firewall adjustment you are already familiar with. Others have told me that NordVPN works as well, but i have not used it myself. It could be worth checking if running up NordVPN (without any other config changes) on your node PC makes the ipfingerprints test report the ports as open. If not, give surfshark a try (they also claim to NOT support port forwarding, but it definitely worked for me 3 months ago)

**Comment by viktorgogulenko** [comment]
I'm using Vodafone ISP in Germany and seems I will not be able to configure Pi Node as even though my ports are opened in router (FritzBox 6490) and on localhost:31400 I'm seeing "OK FROM PORT 31400" , https://www.ipfingerprints.com/portscan.php is still showing that ports are all "filtered" and not opened so Pi Network port listener is failing while checking my ports. Unfortunately seems I'm blocked. :(

**Comment by mokhabarat** [comment]
hey guys, thanks for all ur help. I went the easy way (supposedly) and purchased a vpn subscription called Private Internet Access. they claim to support port forwarding and even have a button in their app to activate it. But somehow it still doesnt work, i contacted their support and they said the problem is with my end. If anyone is familiar with them please advise if there are any special settings to configure in order to get the node working, otherwise ill just return it with the 30 day money back and get surfshark
![privacy tab](https://user-images.githubusercontent.com/15421013/110250990-0bc98200-7f87-11eb-80b4-fe0cb8214f61.jpg)
![connection tab](https://user-images.githubusercontent.com/15421013/110250992-0d934580-7f87-11eb-8191-55c0ea81e961.jpg)
![network tab](https://user-images.githubusercontent.com/15421013/110250993-0e2bdc00-7f87-11eb-9b33-9d66bd5705ff.jpg)


![pi error](https://user-images.githubusercontent.com/15421013/110251176-03257b80-7f88-11eb-9621-a5624ad78214.jpg)

**Comment by ihatejam** [comment]
I don't know this VPN provider -- but try switching your connection type from UDP to TCP and let me know if that's made a difference.
If it has not, let me know if you use Teamviewer or Anydesk?  (do NOT post your Teamviewer or Anydesk details on here)

**Comment by ihatejam** [comment]
Looking at your last pic of the troubleshooting page -- Everything looks fine - your node is working and is in sync. So your port forwarding is working just fine. 
What makes you think that it is not working?

**Comment by mokhabarat** [comment]
i have no incoming connections and i'm not supporting other nodes. is that normal?

**Comment by ihatejam** [comment]
You're fine! 
-- I was going to leave my response at this, but thought you may have more questions, so... 

if your inbound port forwarding was not working, then even establishing "Outgoing connections" would not be possible. 
For most nodes, "Incoming connections=0" is normal -- Installations that are later upgraded to supernode status (based on availability stats) will show more inbound connections - these would automatically switch to "Supporting other nodes=yes".

The stellar-dummy container is the one intitially used when testing all the open ports (31400-31409) - it is automatically stopped when the consensus container is started since it needs access to the same ports. If you did a port test at https://www.ipfingerprints.com/portscan.php while the consensus container is running, it should report only a handful of open ports (as opposed to all ports when the stellar-dummy is running) - this is normal.

hope this helps

**Comment by mokhabarat** [comment]
perfect! thank you for your help

**Comment by cheme75** [comment]
not sure what's going on here on win 10 pc- got docker running with no issues, shows server listening on the ports 31400-31409, stellar-dummy is running. On Pi node info, shows all port green so open - I do "check now" and I get downloading ports listener with green check, starting ports listener with green check, then pinging your computer does nothing. I initially got the node running on an old laptop but it was old and too slow. Left docker and node running in background but too difficult to work on the machine. So, I moved HD with win10 to a less old desktop, got win10 re-authorized, updated all drivers now the docker and node seem to be running reliably. I'm not sure how to get noticed to get the official container and not sure if the ping is a problem hindering my node to be accepted. ipfingerprint port scan shows external ip and shows all ports are open. Kind of frustrating to be sort of stuck in limbo. Any help is appreciated!

**Comment by ihatejam** [comment]
What does https://www.ipfingerprints.com/portscan.php report when you check your port forwarding from a browser on the node pc.

Please don't post pictures of the ipfingerprints results without obfuscating your public IP address.

**Comment by cheme75** [comment]
> What does https://www.ipfingerprints.com/portscan.php report when you check your port forwarding from a browser on the node pc.
> 
> Please don't post pictures of the ipfingerprints results without obfuscating your public IP address.

Says All are Open:

![E103BF4D-5398-4D3C-B934-9920FD851F2D](https://user-images.githubusercontent.com/16415255/148701802-cd951725-73f2-422c-bca9-58c6afb41050.jpeg)

So what’s next?  Too late for node to join network now that main net is going live?  Continued frustration?

**Comment by ihatejam** [comment]
Seems like you're good to go -- just leave the Node in this state until it is picked up later to run the consensus container. Even if the core team does not (for whatever reason) send you the consensus container running against the testnet, your node's listening time will, according to the core team, still be included in the reward calculation mentioned in the last couple of mainnet announcements, just as if it was actually running the consensus container.

**Comment by cheme75** [comment]
> Seems like you're good to go -- just leave the Node in this state until it is picked up later to run the consensus container. Even if the core team does not (for whatever reason) send you the consensus container running against the testnet, your node's listening time will, according to the core team, still be included in the reward calculation mentioned in the last couple of mainnet announcements, just as if it was actually running the consensus container.

Ok. Will give it some time. The old laptop would go to sleep or battery would die if I had to move it from plug so may not have been online continuously enough to get found. This old desktop PC stays online so maybe a better chance of getting picked up. 

Meanwhile, off topic but port related. Some users on various forums question the wisdom of opening ports due to security risks. But with these ports only for incoming traffic, only for this PC, doesn’t that still block sending data on these ports?  Why would these open ports be any more of a risk than port 80 for web traffic or 9100 for printer access or the various ports for email, for example?

**Comment by ihatejam** [comment]
Yep - not much risk if you're forwarding ports to docker container does not have access to anything meaningful on your PC.

**Comment by cheme75** [comment]
FWIW, it just took some time - I left it running and the consensus container loaded and is running well.

**Comment by zlove304** [comment]
I dont have access to my routers admin page because my isp locks it down 
So i went with PIA vpn enabled port forwarding but still no luck.
looking for any help thanks
![SharedScreenshot](https://user-images.githubusercontent.com/104103609/164326964-a89bab03-48c2-4d72-b8e6-19ced84dbb85.jpg)
![SharedScreenshot2](https://user-images.githubusercontent.com/104103609/164326966-fc6fa53b-97d9-4fd2-97fb-39ebc00e028d.jpg)

**Comment by AbelKinkela** [comment]
Guys I am trying to use my phone's hotspot. And getting ping computer all ports not open. Any chances this will work?

**Comment by cheme75** [comment]
> Guys I am trying to use my phone's hotspot. And getting ping computer all ports not open. Any chances this will work?

Read further up, you’ll probably need to use a vpn to open the ports. Although if you are pinging from computer, you might just need to open the ports on the pc firewall. I added a rule to allow incoming traffic for the ports to my docker pc ip, from any ip. Also I did have to port fwd each port in my router to the docker pc ip - not sure if you can do same via vpn.

**Comment by abebawmulat** [comment]
I was trying to run a pi node on my pc. I try to open a port from 31400-31409 on D-Link port. And also, I adjust the setting on the firewall to have a permission for those ports. But after following those steps, even now I can't run the pi node on my pc. Can any one help me?
![blockchain](https://user-images.githubusercontent.com/88592417/193335731-4e987c37-171b-4238-97a9-19a32386b36b.jpg)
![D-link](https://user-images.githubusercontent.com/88592417/193335748-e81b1cda-7339-439d-943c-b5de906d2197.jpg)
![firewall](https://user-images.githubusercontent.com/88592417/193335773-58e57185-2201-45b5-b067-0ec23e3caa85.jpg)
![port problem](https://user-images.githubusercontent.com/88592417/193335787-fa2ce0f1-ab3a-4737-8adf-739308024778.jpg)

**Comment by cheme75** [comment]
not sure what the service AUTH is used for on the system, but rather than use a pre-defined service and change its settings - I setup a custom or user defined service "pi-node" then setup its port forwarding - I don't think it matters unless AUTH was a predefined service that was in-use that is now changed and not working - I guess if all other networking seems normal, its ok. Otherwise things look ok since all the ports are in green, I believe if they are found closed or unavailable, they are listed in red. For me, it took several tries to get the dummy container which is used for the ports checker - I found some info in Issue #93 that helped get both the ports checker to finish and then get the consensus container for the testnet - I'm stil without kyc so its still on testnet, but it seems to be ok most of the time, although the mobile app can go days without any bonus.

**Comment by mrtaotao789** [comment]
All port are OPEN via the other not the PI-NODE, i very confuse T_T
Someone please help me or remote to config my com.Thanks.

![Screen Shot 10-01-22 at 09 00 AM](https://user-images.githubusercontent.com/31775186/193379679-7309497a-b32f-47ff-9c9d-f3fc802fe204.PNG)
![Screen Shot 10-01-22 at 08 59 AM](https://user-images.githubusercontent.com/31775186/193379682-d62bdf6d-bec0-4569-91ee-e375acb72272.PNG)
![Screen Shot 10-01-22 at 09 29 AM](https://user-images.githubusercontent.com/31775186/193379718-e88138c1-5f23-4978-b43d-3d9247fa7b8b.PNG)

**Comment by cheme75** [comment]
Does your docker dashboard show stellar-dummy in the containers section? If not there, something may be blocking docker from pulling that container which is what does the port checking. See issue #93  for info on things to do to be sure the ports are setup right on the pc, and then reinstall docker. Worked for me.

**Comment by mrtaotao789** [comment]
now Pi-Node don t work, i don t understand,
Waht ip PI-NODE checked and how do i config that ,
Please someone remote to fix, Thanks T_T
![Screen Shot 10-01-22 at 09 00 AM](https://user-images.githubusercontent.com/31775186/193391089-d95275fb-301a-4993-940f-0e32ac6dfec5.PNG)
![Screen Shot 10-01-22 at 08 59 AM](https://user-images.githubusercontent.com/31775186/193391122-0d76076f-9d77-43e6-9235-924ccef561ec.PNG)

**Comment by mrtaotao789** [comment]
> Does your docker dashboard show stellar-dummy in the containers section? If not there, something may be blocking docker from pulling that container which is what does the port checking. See issue #93 for info on things to do to be sure the ports are setup right on the pc, and then reinstall docker. Worked for me.

Yes, it has stellar-dummy in the containers section, when press button Check-now in PI-NODE App, it run normal but all port -fw are close. Does you remote via Anydesk-App remote to fix this, thanks

for #93
i followed all step, result all port are close.

![Screen Shot 10-01-22 at 04 06 PM](https://user-images.githubusercontent.com/31775186/193402063-e1d3082a-0dbc-4c50-9f5e-d4d21276649b.PNG)

**Comment by mrtaotao789** [comment]
For this pic, i press Play button on Containers-Docker and check port-fw via WEBSITE and PI-NODE in same time.https://www.ipfingerprints.com/portscan.php 
How do i fix it, wait for someone remote to fix that, thanks
![Screen Shot 10-01-22 at 04 28 PM](https://user-images.githubusercontent.com/31775186/193403153-0138e316-455a-4aad-bf13-2b5e6c2a54cc.PNG)

**Comment by NeyMR** [comment]
hello 👋, need help?? , join us in our Node help chat https://t.me/PiNodechat , we're always glad to help 🤓👍

**Comment by cheme75** [comment]
Try the node chat. I don’t use that app. Best I can tell, it may be be your windows firewall - try creating rule for in and out. Plus docker sets up a separate subnet that may need a rule, my normal network is 192.168.x.x, but docker is on 172.x.x.x - may need to allow tcp traffic on that subnet.

**Comment by cheme75** [comment]
> For this pic, i press Play button on Containers-Docker and check port-fw via WEBSITE and PI-NODE in same time.https://www.ipfingerprints.com/portscan.php
> How do i fix it, wait for someone remote to fix that, thanks
> ![Screen Shot 10-01-22 at 04 28 PM](https://user-images.githubusercontent.com/31775186/193403153-0138e316-455a-4aad-bf13-2b5e6c2a54cc.PNG)

I believe checking ports sends the run command to the container. I never had to start the stellar-dummy separately. After I got it to check my ports I just left everything alone for a week or so after I got the pink button to run the blockchain and it eventually got the consensus container and started running.

**Comment by mrtaotao789** [comment]
> Try the node chat. I don’t use that app. Best I can tell, it may be be your windows firewall - try creating rule for in and out. Plus docker sets up a separate subnet that may need a rule, my normal network is 192.168.x.x, but docker is on 172.x.x.x - may need to allow tcp traffic on that subnet.

now i close all Firewall (close.. Private Public and Domain) for test, result as... my port forward are open via test on website (https://www.ipfingerprints.com/portscan.php) but not-open via pi-node.

i have 2 pc and now they are same result T_T!

**Comment by Donnie1992** [comment]
Same problem here, port scan and everything is open.
Yet still shows closed in checker. After that it also writes portsCheckPassed":true into the user preferences.json file.
So when u open it again it just gets stuck on loading.

**Comment by N0JDAR** [comment]
Can anyone help me set a new port in the Huawei HG8245H router because I'm tired of trying, specifically port 31400-31409
![image](https://user-images.githubusercontent.com/131220844/232945886-235b3cec-a4ab-45f9-8fa9-7bc6357f3f73.png)

**Comment by BittusainiBS** [comment]
> I really appreciate the detailed help THANK YOU!
> 
> > fyi - your LAN IP address cannot be used by us outside of your network - so it is typically not necessary to mask it (but it doesn't hurt to be careful).
> 
> **Didn't know this**
> 
> > It is the router's public/WAN ip address that needs to be masked - this is the one that is typically directly reachable from outside - which I in fact suspect is not the case here and is the source of your grief...
> > The thing to check next is whether the ISP is doing their part - i.e. allowing port-forwarding. Something that could help you check this is:-
> > (1.) go here to check your port forwarding - it will autoreport your public IP address- chances are that it will fail - but it helps to sanity check things - https://www.ipfingerprints.com/portscan.php
> > (1.1.) are ports 31400-31409 reporting open? (I'm guessing not) Host is up.
> 
> **PORT - STATE - SERVICE**
> 
> **31400/tcp filtered pace-licensed**
> 
> **others are "filtered" state with "unknown" service**
> 
> > (1.2.) note down the public ip address that this site reports you to be using (do NOT post it here) - I'll call it <PUBLIC_IP>
> > (2) Check the config of your router to see what ip address it reports as being assigned by your ISP (do NOT post it here) - let's call it - note it down.
> > If <PUBLIC_IP> and are the same, then your setup/config is likely the issue (if the ISP is not blocking port-forwarding).
> > Now if <PUBLIC_IP> and are NOT the same, then either
> > (A) you have an additional router (sometimes more than one) sitting between your main router (usually provided by your ISP) and the router you are configuring the forwarding on.
> > or
> > (B) your ISP does not provide "true" public IP addresses to its subscribers - so your port-forwarding efforts will have zero effect.
> 
> **It seams to be different (Modem IP Address is what I'm looking at correct?)** ![settings](https://user-images.githubusercontent.com/59236513/109343531-c2609080-782a-11eb-9620-147f5eb5418f.png)
> 
> > To recover from (A) a bit more work is needed - you need to forward those same ports from the ISP's router to the router connecting to your node. If there are multiple routers in between then each will have to be set up with forwarding those ports to the next router in the chain. You can also get around this with a VPN service (supporting port-forwarding) like Surfshark or NordVPN.
> > To recover from (B) with help from your ISP - call them up and explain that you want to do port forwarding. If they can't help - then either move to an ISP that can do it or to use a VPN service (supporting port-forwarding) like Surfshark or NordVPN.
> 
> **Ill call my ISP but I have the feeling that it wont make a difference based off forums where they're mentioned.**
> 
> **I have Nord, I didn't know it could do that. I'm guessing its the custom DNS or manually changing the "TCP/IPv4" to the** **NordVPN's DNS server. Otherwise on their FAQ port forwarding isn't integrated for safety.**
> 
> **Once again thank you for replying**



> DHCP Reservation - router forwarding - firewall input rule
> 
> First - DHCP Reservation for my computer with the Mac address associated with my ethernet port. Second - Through the router with the reserved IP I've done a port forwarding with the ports 31400-31409. Third - Made a new input rule in windows following the instructions 31400-31409.
> 
> ![Pi node](https://user-images.githubusercontent.com/59236513/109334998-ab1ba600-781e-11eb-9133-3083d22aa784.png)
> 
> Port forwarding w/ view of my available router firewall and advanced options. ![T3200m](https://user-images.githubusercontent.com/59236513/109336630-eb7c2380-7820-11eb-8020-6f8ac0e91b6e.png)

hay please help me

**Comment by BittusainiBS** [comment]
i have tried everything but still my 31400-31409 ports are closed
